src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_help.py

```diff
@@ -22,6 +22,25 @@
           text: az {1} -n name -g MyResourceGroup
 """
 
+helps['vm format-secret'] = """
+    type: command
+    long-summary: >
+        Transform secrets into a form consumed by VMs and VMSS create via --secrets.
+    examples:
+        - name: Create a self-signed certificate with a the default policy and add to a virtual machine
+          text: >
+            az keyvault certificate create --vault-name vaultname -n cert1 \\
+              -p "$(az keyvault certificate get-default-policy)"
+
+            secrets=$(az keyvault secret list-versions --vault-name vaultname \\
+              -n cert1 --query "[?attributes.enabled]")
+
+            vm_secrets=$(az vm format-secret -s "$secrets") \n
+
+            az vm create -g group-name -n vm-name --admin-username deploy  \\
+              --image debian --secrets "$vm_secrets"
```

Comment:
To recap our f2f:
- Make `--secrets` on VM create support `nargs='+', type=json.loads` and have VM create do the work of merging the different outputs from `az vm format-secrets`. Keep the output of `format-secrets` as that required by VM create (not a simplified intermediate one).
- Have the `vm format-secrets --secrets` arg take a list of secret URIs rather than the JSON output of the list command (or, have it accept both). You'd have to use query and TSV output to feed this in, but that's consistent with our current `--ids` usage elsewhere in the CLI.